nilearn/plotting/tests/test_matplotlib_backend.py

```diff
@@ -0,0 +1,46 @@
+from unittest.mock import patch
+
+SKIP_REASON = "matplotlib missing; necessary for this test"
+try:
+    import matplotlib  # noqa: F401
+except ImportError:
+    MATPLOTLIB_INSTALLED = False
+else:
+    MATPLOTLIB_INSTALLED = True
+import pytest
+
+from nilearn.plotting import _set_mpl_backend
+
+
+@pytest.mark.skipif(not MATPLOTLIB_INSTALLED, reason=SKIP_REASON)
+@patch("matplotlib.use")
+@patch("matplotlib.get_backend", side_effect=["backend_1", "backend_2"])
+def test_should_raise_warning_if_backend_changes(*_):
+    # The backend values returned by matplotlib.get_backend are different.
+    # Warning should be raised to inform user of the backend switch.
+    with pytest.warns(UserWarning, match="Backend changed to backend_2..."):
+        _set_mpl_backend()
+
+
+@pytest.mark.skipif(not MATPLOTLIB_INSTALLED, reason=SKIP_REASON)
+@patch("matplotlib.use")
+@patch("matplotlib.get_backend",
+       side_effect=["backend_1", "backend_1"])
+def test_should_not_raise_warning_if_backend_is_not_changed(*_):
+    # The backend values returned by matplotlib.get_backend are identical.
+    # Warning should not be raised.
+    with pytest.warns(None):
+        _set_mpl_backend()
+
+
+@pytest.mark.skipif(not MATPLOTLIB_INSTALLED, reason=SKIP_REASON)
+@patch("matplotlib.use",
+       side_effect=[Exception("Failed to switch backend"), True])
+def test_should_switch_to_agg_backend_if_current_backend_fails(use_mock):
+    # First call to `matplotlib.use` raises an exception, hence the default Agg
+    # backend should be triggered
+    _set_mpl_backend()
```

Comment:
here also you could catch the warning

Reply:
I don't think so, because if the backend used during the test is already `Agg`, the warning will not be raised. Hence, I'd rather not test the warning here as it could lead to random fails + it is tested on its own

Reply:
I meant suppress the warning to avoid having it pollute pytest's warning summary. but actually it doesn't show up because:
- conftest.py specifies that we always use agg in tests
- and even if that wasn't the case, use is mocked but not get_backend so we never have new_backend != current_backend

so we can keep it as it is